tests/test_linear_objectives.py

```diff
@@ -998,13 +996,11 @@ def test_fix_coil_current(DummyMixedCoilSet):
     # fix all coil currents
     obj = FixCoilCurrent(coil=coilset)
     obj.build()
-    assert obj.dim_f == 8
-    np.testing.assert_allclose(obj.target, [3, 3, 3, 3, -1, -1, -1, 2])
+    assert obj.dim_f == 6
+    np.testing.assert_allclose(obj.target, [3, -1, -1, -1, 2, 1])
 
     # only fix currents of some coils in the coil set
-    obj = FixCoilCurrent(
-        coil=coilset, indices=[[True, False, True, False], False, True]
-    )
+    obj = FixCoilCurrent(coil=coilset, indices=[[True], False, True, True])
```

Comment:
It would be good to still include a case where only some coils within a `CoilSet` have fixed currents. Now only the vertical field coilset has multiple coils, so I think something like this should work: 
```
obj = FixCoilCurrent(coil=coilset, indices=[[True], [True, False, True], True, False])
```
and then I think the test in the line below would be: 
```
np.testing.assert_allclose(obj.target, [3, -1, -1, 2])
```